tasks.c

```diff
@@ -6897,13 +6974,44 @@ static void prvResetNextTaskUnblockTime( void )
                         break;
                 }
 
-                /* Write the task name to the string, padding with spaces so it
-                 * can be printed in tabular form more easily. */
-                pcWriteBuffer = prvWriteNameToBuffer( pcWriteBuffer, pxTaskStatusArray[ x ].pcTaskName );
+                /* Is there enough space in the buffer to hold task name? */
+                if( ( uxConsumedBufferLength + configMAX_TASK_NAME_LEN ) <= uxBufferLength )
+                {
+                    /* Write the task name to the string, padding with spaces so it
+                     * can be printed in tabular form more easily. */
+                    pcWriteBuffer = prvWriteNameToBuffer( pcWriteBuffer, pxTaskStatusArray[ x ].pcTaskName );
+                    uxConsumedBufferLength += configMAX_TASK_NAME_LEN;
+                }
+                else
+                {
+                    /* Output buffer full. */
+                    break;
```

Comment:
MISRA 2012 rule 15.4 suggests that "There should be no more than one break or goto statement used to terminate any iteration statement". This rule is of category advisory. Can we consider to use one break in the loop only?